### PR TITLE
Custom chat component

### LINKED ISSUE #75: Implement Chat Feature

### PR DESCRIPTION
**Pull request by dani5447**
Closes #75.
Does not include chat history and chat messages don't persist across pages.

### CODE REVIEW

**Comment by dani5447** [comment]
image

**Comment by bbesmanoff** [comment]
Can you make tickets for the remaining work from the PR description?

**Comment by dani5447** [comment]
Done

File: app/js/components/chat.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+var that;
+
+export default class Chat extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            messages:[{
+                timeStamp: Date.now(),
+                user: 'SYSTEM',
+				text: "Welcome to the chat!",
```
Comment (bbesmanoff) [inline app/js/components/chat.js]: NP: Formatting

File: app/js/components/chat.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        message.received = true;
+		that.setState({messages: that.state.messages.concat(message)});
+	}
+    
+    getCookieValue(cname){
+        var name = cname + "=";
+        var ca = document.cookie.split(';');
+        for(var i=0; i<ca.length; i++) {
+            var c = ca[i];
+            while (c.charAt(0)==' ') c = c.substring(1);
```
Comment (bbesmanoff) [inline app/js/components/chat.js]: `c.trim()`

File: server.js [hunk 3]
```diff
@@ -105,4 +107,20 @@ if (server.get('env') !== 'production') {
 
 server.use('/api', api);
 
-export default server;
+//Chat server 
+var chatServer = http.createServer(server);
+var io = socketio.listen(chatServer);
+// server events
+io.on('connection', function(socket){
+	console.log('a user connected');
```
Comment (bbesmanoff) [inline server.js]: Stray `console.log`(s)?

**Comment by bbesmanoff** [comment]
A few comments inline.  Also (NP), several files had mixed tabs/spaces which should probably be cleared up

**Comment by bbesmanoff** [comment]
:+1:

**Pull request merged**